### PR TITLE
add js-sha1@0.3.0 w/ npm auto-update

### PR DESCRIPTION
**Pull request by nolsherry**
@pvnr0082t, please help me review this PR, thanks!
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/emn178/js-sha1
 * Official website (optional, not the repository): 
 * NPM package url (optional): https://www.npmjs.com/package/js-sha1
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 12
   - Count of watchers: 2
   - Count of forks: 6
 * NPM download stats (optional):
   - Downloads in the last day: 64
   - Downloads in the last week: 547
   - Downloads in the last month: 2286

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 500 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

cc @emn178

### CODE REVIEW

**Comment by pvnr0082t** [comment]
@nolsherry There is officially minified file under `build` folder. You don't need to minify yourself.

**Comment by nolsherry** [comment]
@pvnr0082t, I have modified the `package.json`, and updated with npm auto-update config.
Please help me review this PR, thanks.

File: ajax/libs/js-sha1/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "basePath": "src",
+      "files": [
+        "sha1*"
+      ]
+    },
+    {
+      "basePath": "build",
+      "files": [
+        "sha1.min.js"
```
Comment (pvnr0082t) [inline ajax/libs/js-sha1/package.json]: We can use `**/*` here.

**Comment by nolsherry** [comment]
@pvnr0082t, I have modified `package.json`, please help me review this PR, thanks.

**Comment by pvnr0082t** [comment]
LGTM

**Comment by pvnr0082t** [comment]
@nolsherry Don't forget to assign to reviewer.

**Comment by PeterDaveHello** [comment]
LGTM, @pvnr0082t please try to merge this!

**Pull request merged**